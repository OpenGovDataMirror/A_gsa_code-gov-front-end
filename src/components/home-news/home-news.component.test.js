import React from 'react'
import { shallow } from 'enzyme'
import HomeNews from 'components/home-news/home-news.component'

const props = {
  newsItems: [
    { title: 't1', url: 'u1', description: 'd1', date: 'dt1', image: 'i1', alt: 'alt1' },
    { title: 't2', url: 'u2', description: 'd2', date: 'dt2', image: 'i2', alt: 'alt2' },
    { title: 't3', url: 'u3', description: 'd3', date: 'dt3', image: 'i3', alt: 'alt3' }
  ]
}

let wrapper

describe('components - HomeNews', () => {
  beforeEach(() => {
    wrapper = shallow(<HomeNews {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map all the `exploreItems`', () => {
    expect(wrapper.find('li').length).toBe(props.newsItems.length)
  })
})
