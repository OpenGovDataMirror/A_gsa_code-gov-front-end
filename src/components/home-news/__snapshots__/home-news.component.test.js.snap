// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomeNews should render correctly 1`] = `
<section
  className="tablet:grid-col-6"
>
  <div
    className="bg-primary-lighter"
  >
    <header
      className="padding-x-4  padding-top-6"
    >
      <h2
        className="font-heading-l text-primary-darker text-center"
      >
        Open Source News & Events
      </h2>
    </header>
    <ul
      className="padding-x-1 padding-bottom-2"
    >
      <li
        className="bg-white grid-row margin-1 padding-2"
        key="t1"
      >
        <div
          className="flex-1"
        >
          <h3
            className="font-heading-sm text-bold"
          >
            <a
              href="u1"
              rel="noopener noreferrer"
              target="_blank"
            >
              t1
            </a>
          </h3>
          <p
            className="margin-bottom-0 font-body-2xs margin-top-105"
          >
            d1
          </p>
        </div>
        <picture
          className="order-first padding-right-2"
        >
          <source
            media="min-width: 800px"
            srcSet="i1"
          />
          <img
            alt="alt1"
            className="maxw-15 margin-right-1"
            src="i1"
          />
        </picture>
      </li>
      <li
        className="bg-white grid-row margin-1 padding-2"
        key="t2"
      >
        <div
          className="flex-1"
        >
          <h3
            className="font-heading-sm text-bold"
          >
            <a
              href="u2"
              rel="noopener noreferrer"
              target="_blank"
            >
              t2
            </a>
          </h3>
          <p
            className="margin-bottom-0 font-body-2xs margin-top-105"
          >
            d2
          </p>
        </div>
        <picture
          className="order-first padding-right-2"
        >
          <source
            media="min-width: 800px"
            srcSet="i2"
          />
          <img
            alt="alt2"
            className="maxw-15 margin-right-1"
            src="i2"
          />
        </picture>
      </li>
      <li
        className="bg-white grid-row margin-1 padding-2"
        key="t3"
      >
        <div
          className="flex-1"
        >
          <h3
            className="font-heading-sm text-bold"
          >
            <a
              href="u3"
              rel="noopener noreferrer"
              target="_blank"
            >
              t3
            </a>
          </h3>
          <p
            className="margin-bottom-0 font-body-2xs margin-top-105"
          >
            d3
          </p>
        </div>
        <picture
          className="order-first padding-right-2"
        >
          <source
            media="min-width: 800px"
            srcSet="i3"
          />
          <img
            alt="alt3"
            className="maxw-15 margin-right-1"
            src="i3"
          />
        </picture>
      </li>
    </ul>
  </div>
</section>
`;
